src/rollup/types.d.ts

```diff
@@ -74,11 +74,10 @@ export interface ExistingRawSourceMap {
 
 export type DecodedSourceMapOrMissing =
 	| {
-			mappings?: never;
 			missing: true;
 			plugin: string;
 	  }
-	| ExistingDecodedSourceMap;
+	| (ExistingDecodedSourceMap & { missing?: false });
```

Comment:
This change was to avoid using `!map.mappings` to determine if the map was missing as accessing this property would trigger the decode.

With this change, the only option provided by intellisense is `.missing` and switching on that value correctly narrows the type using control flow analysis.